### PR TITLE
Feature for auto-publishing web pages based on their start and end dates

### PR DESCRIPTION
**Pull request by Alchez**
Scenarios accounted for:
* **No Start and End date:** User can change publication status in any combination.
* **Start date, but no End date:** User can change publication status in any combination.
* **End date, but no Start date:** System automatically unpublishes Web Page when End date is passed. If user manually publishes the page, the End Date blanks out.
* **Both Start and End date:**
  * **Publish:** System automatically unpublishes Web Page when End date is passed. If user manually publishes the page, the End Date blanks out.
  * **Unpublish:** If user unpublishes the Web Page, End date changes to current datetime.

publish

### CODE REVIEW

File: frappe/website/doctype/web_page/web_page.py [hunk 1]
```diff
@@ -121,6 +130,34 @@ def set_metatags(self, context):
 		if image:
 			context.metatags["image"] = image
 
+	def check_publication_dates(self):
+		if self.valid_upto and self.valid_upto < self.valid_from:
```
Comment (vjFaLk) [inline frappe/website/doctype/web_page/web_page.py]: Change valid_* to start_ / end_

**Review by vjFaLk — APPROVED** [review]
I have no idea what I'm doing.

**Comment by vjFaLk** [comment]
@Alchez Let's merge this into the core.

**Comment by Alchez** [comment]
PR created to core at https://github.com/frappe/frappe/pull/5430.

**Pull request closed without merging**